pkg/resource/plugin/analyzer_plugin.go

```diff
@@ -305,3 +313,46 @@ func convertDiagnostics(protoDiagnostics []*pulumirpc.AnalyzeDiagnostic) ([]Anal
 
 	return diagnostics, nil
 }
+
+func constructEnv(opts *PolicyAnalyzerOptions) ([]string, error) {
+	env := os.Environ()
+
+	maybeAppendEnv := func(k, v string) {
+		if v != "" {
+			env = append(env, k+"="+v)
+		}
+	}
+
+	config, err := constructConfig(opts)
+	if err != nil {
+		return nil, err
+	}
+	maybeAppendEnv("PULUMI_CONFIG", config)
+
+	if opts != nil {
+		maybeAppendEnv("PULUMI_NODEJS_PROJECT", opts.Project)
+		maybeAppendEnv("PULUMI_NODEJS_STACK", opts.Stack)
+		maybeAppendEnv("PULUMI_NODEJS_DRY_RUN", fmt.Sprintf("%v", opts.DryRun))
+	}
+
+	return env, nil
+}
+
+// constructConfig json-serializes the configuration data.
+func constructConfig(opts *PolicyAnalyzerOptions) (string, error) {
```

Comment:
Pedantic-nit: JSON is an acronym, so all caps.

Reply:
Sure. We have existing variants of this code for each of the language hosts, for example:

https://github.com/pulumi/pulumi/blob/048acc24f774fa51916d418c45ccc31a05f3f250/sdk/dotnet/cmd/pulumi-language-dotnet/main.go#L512-L548